core/src/main/java/hudson/tasks/BuildTrigger.java

```diff
@@ -151,21 +153,31 @@ public Result getThreshold() {
         return getChildProjects(Jenkins.getInstance());
     }
 
+    /** @deprecated use {@link #getChildJobs} */
+    @Deprecated
     public List<AbstractProject> getChildProjects(AbstractProject owner) {
         return getChildProjects(owner==null?null:owner.getParent());
     }
 
+    @Deprecated
     public List<AbstractProject> getChildProjects(ItemGroup base) {
         return Items.fromNameList(base,childProjects,AbstractProject.class);
     }
 
+    @SuppressWarnings("unchecked")
+    @Nonnull
+    public List<Job<?, ?>> getChildJobs(@Nonnull AbstractProject<?, ?> owner) {
```

Comment:
To @oleg-nenashev’s comment (made on a commit rather than in the PR):

We have to—only `AbstractProject` supports `DependencyGraph`.